nix/process-compose/cli-options.nix

```diff
@@ -0,0 +1,145 @@
+{ lib }:
+let
+  inherit (lib) types mkOption;
+in
+lib.mkOption {
+  type = types.submodule {
+    options = {
+      global = lib.mkOption {
+        type = types.submodule ({ config, ... }: {
+          options = {
+            log-file = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+            };
+            no-server = mkOption {
+              type = types.bool;
+              default = false;
+            };
+            ordered-shutdown = mkOption {
+              type = types.bool;
+              default = false;
+            };
+            port = mkOption {
+              type = types.nullOr types.int;
+              default = null;
+            };
+            read-only = mkOption {
+              type = types.bool;
+              default = false;
+            };
+            unix-socket = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+            };
+            use-uds = mkOption {
+              type = types.bool;
+              default = false;
+            };
+            output = mkOption {
+              type = types.str;
+              internal = true;
+              default = "";
+            };
```

Comment:
I was confused by this option. `output` is nothing like the rest of the options in this group, and as such it should be moved elsewhere in the hierarchy. 

```nix
{
  global = { ... }
  up = { ... }
  cliArguments = { global = ".."; up = ".."; }
}
```

Also, instead of `default = ""` (which is meaningless) you want to assign the implementation directly here, instead of defining it in `config = {` below.

Reply:
It's there because its value is computed from the other options of that group, which makes sense. It also made it easy go copy the whole option tree, but since we're removing the copy(test-cli) anyway, that's not a valid reasoning anymore.

Reply:
Regarding the default="", i think this has also to do with the "input" options being in the same submodule. The default was completely ignored in my tests as soon as i'se set one of the input options from elsewhere. But again, i'm no expert, it's just something i noticed duri g tinkering.